src/Serilog.Sinks.Fluentd/Sinks/Fluentd/FluentdSinkClient.cs

```diff
@@ -21,33 +21,32 @@ public FluentdSinkClient(FluentdSinkOptions options)
             _options = options;
         }
 
-        protected void InitializeTcpClient()
+        protected void InitializeEndpoint()
         {
             Cleanup();
 
-            _tcpClient = new TcpClient
+            if(_options.UseUnixDomainSocketEndpoit)
             {
-                NoDelay = _options.NoDelay,
-                ReceiveBufferSize = _options.ReceiveBufferSize,
-                SendBufferSize = _options.SendBufferSize,
-                SendTimeout = _options.SendTimeout,
-                ReceiveTimeout = _options.SendTimeout,
-                LingerState = new LingerOption(_options.LingerEnabled, _options.LingerTime)
-            };
-
+                _endpoint = new UdsEndpoint(_options);
+            }
+            else
+            {
+                _endpoint = new TcpEndpoint(_options);
+            }
         }
 
         protected async Task EnsureConnectedAsync()
         {
             try
             {
-                if (IsConnected()) return;
+                bool _endpointInitialzied = _endpoint?.IsConnected() ?? false;
+                if(_endpointInitialzied) return;
```

Comment:
```
if(_endpointInitialzied) {
    return;
}
```
It is not valuable, but looks more cleaner.

Reply:
Good point, thanks! 🙂  Don't know why did I put and underscore there, probably forgot to remove it when I was renaming the variable. 🤦‍♂️